wp/sections/compare.tex

```diff
@@ -0,0 +1,48 @@
+\subsection{About ktlint}
+\par Ktlint is a popular an anti-bikeshedding Kotlin linter with built-in formatter. It tries to capture (reflect) official code style from kotlinlang.org and Android Kotlin Style Guide and then automatically apply these rules to your codebase. It claims to be simple and easy to use. Ktlint has been developing since 2016 and from then on it has 3.8k stars, 299 forks and 390 closed PRs (at least on the moment of writing this whitepaper). There have been written over 15k lines of code. Ktlint has it's own ruleset, which divides on standard and experimental rules. Ktlint can be used as a plugin via Maven or Gradle. To configure rules in Ktlint you should modify .editorconfig file. But you can't configure specific rules, instead you can provide some common settings. Properties should be specified under \textbf{[*.{kt,kts}]}. If you want to implement your own rules you need to create a ruleset. Ktlint is using java's ServiceLoader to discover all available "RuleSets". Basically you should provide implementation of "RuleSetProvider" interface. Ktlint refers to article on \textbf{medium} \footnote{\url{https://medium.com/@vanniktech/writing-your-first-ktlint-rule-5a1707f4ca5b}}on how to create a ruleset and a rule.
```

Comment:
write more about Service Loader, something like this: "ServiceLoader is used to inject your own implementation of rules for the static analysis. In this case ktlint becomes a third-party dependency and a framework."